### PR TITLE
Feat/25 add catalog info

### LINKED ISSUE #25: Add catalog-info.yaml so project can be registered to the arcoplex-portal upon creation
The default workflow will be to create projects from the arcoplex-portal. It would be ideal if the project could be registered when it was created. This requires the project to have a catalog-info.yaml file. Most of the information in the catalog could be filled in by cookiecutter.

https://backstage.io/docs/features/software-catalog/descriptor-format

### PR DESCRIPTION
**Pull request by slougheed**
Added catalog_info.yaml so new projects can be registered with the arcoplex-portal
Added a CHANGELOG for the template itself.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+### Added/Removed/Changed/Deprecated/Fixed/Security
+```
+
+## [Latest Changes]
+
+## [0.2.0] - 2022-03-18
+
+### Added
+
+- catalog-info.yaml so the project can be registered with teh arcoplex-portal
```
Comment (lucasmelin) [inline CHANGELOG.md:21]: Typo.
```suggestion
- catalog-info.yaml so the project can be registered with the arcoplex-portal
```

File: {{cookiecutter.hyphenated}}/catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: {{ cookiecutter.hyphenated }}
+  title: {{ cookiecutter.project_name }}
+
+  description: {{ cookiecutter.project_short_description}}
```
Comment (lucasmelin) [inline {{cookiecutter.hyphenated}}/catalog-info.yaml:7]: Nit - the spacing of `}}` on this line looks different from the rest. Probably doesn't impact anything, but it caught my eye.

**Review by lucasmelin — APPROVED** [review]
I have a few minor comments, but still LGTM. 📦

File: {{cookiecutter.hyphenated}}/catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  annotations:
+    github.com/project-slug: {{ cookiecutter.hyphenated }}
+    backstage.io/techdocs-ref: dir:.
+spec:
+  type: service
+  owner: dev
```
[the remark this answers is not in the document]
Comment (slougheed) [reply {{cookiecutter.hyphenated}}/catalog-info.yaml:13]: Definitely not, good catch. This does pose a slight problem though as the owner field is required but it's not something we currently pass in.

File: {{cookiecutter.hyphenated}}/catalog-info.yaml (hunk 2 above)
Comment (slougheed) [reply {{cookiecutter.hyphenated}}/catalog-info.yaml:7]: That would bother me as well, I'll correct it.

File: cookiecutter.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
 	"full_name": "First Last",
 	"email": "user@bank-banque-canada.ca",
+	"github_owner": "team or username",
```
Comment (lucasmelin) [inline cookiecutter.json:4]: 👍🏻

**Pull request merged**